### PR TITLE
added leave method

### PR DESCRIPTION
**Pull request by Timovanmilligen**
please review

### CODE REVIEW

File: src/main/java/org/distributed/systems/chord/messaging/LeaveMessage.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.distributed.systems.chord.models.ChordNode;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public class LeaveMessage {
+    public static class ForPredecessor implements Command {
+        private ChordNode successor;
+    public ForPredecessor(ChordNode successor){
```
Comment (stienis96) [inline src/main/java/org/distributed/systems/chord/messaging/LeaveMessage.java:12]: formatting?

File: src/main/java/org/distributed/systems/chord/actors/StorageActor.java [hunk 2]
```diff
@@ -36,6 +37,10 @@ public Receive createReceive() {
                     Serializable val = this.storageService.get(getValueMessage.key);
                     getContext().getSender().tell(new KeyValue.GetReply(val), ActorRef.noSender());
                 })
+                .match(KeyValue.GetAll.class, getAllMessage->{
+                    Map<String,Serializable> keyValues = this.storageService.getAll();
```
Comment (stienis96) [inline src/main/java/org/distributed/systems/chord/actors/StorageActor.java]: Should we get all or just the ones that are set?

**Review by stienis96 — APPROVED** [review]
Okay to merge, just some comments and needs to fix the merge conflicts

**Comment by stienis96** [comment]
Was this btw tested with running the jar from the cli and then stopping it?

With these commands for example:
`mvn clean packge -DskipTests`
`cd target`
`java -Dconfig.resource=/centralNode.conf -jar chord-1.0-allinone.jar`
`ctrl + c`

File: src/main/java/org/distributed/systems/chord/actors/StorageActor.java (hunk 2 above)
Comment (Timovanmilligen) [reply src/main/java/org/distributed/systems/chord/actors/StorageActor.java]: you just retrieve all key values, they all need to be passed to your successor

**Comment by Timovanmilligen** [comment]
> Was this btw tested with running the jar from the cli and then stopping it?
> 
> With these commands for example:
> `mvn clean packge -DskipTests`
> `cd target`
> `java -Dconfig.resource=/centralNode.conf -jar chord-1.0-allinone.jar`
> `ctrl + c`

the clean package part is giving me an error, can someone else test this perhaps

**Comment by stienis96** [comment]
Sorry typo in package.. Should work like this if you're in the root dir of the project: `mvn clean package -DskipTests`

**Pull request merged**